autoendpoint/src/extractors/subscription.rs

```diff
@@ -308,7 +337,34 @@ fn validate_vapid_jwt(
                 return Err(VapidError::InvalidVapid(e.to_string()).into());
             }
             _ => {
-                metrics.clone().incr("notification.auth.bad_vapid.other");
+                // Attempt to match up the majority of ErrorKind variants.
+                // The third-party errors all defer to the source, so we can
+                // use that to differentiate for actual errors.
+                let mut tags = Tags::default();
+                let label = if e.source().is_none() {
+                    // These two have the most cardinality, so we need to handle
+                    // them separately.
```

Comment:
I don't understand this comment and the other mention of cardinality a few lines below -- there's some sense of cardinality across all these ErrorKind variant since there's almost 20 of them, but the concern with these 2 variants and others is that their `to_string` includes whitespace which aren't valid for tag values

Reply:
I'm guessing that this probably refers to a different line, since the reference now points to where we extract the raw claims string from the VAPID token.